addons/website_hr_recruitment/static/src/js/website_hr_applicant_form.js

```diff
@@ -0,0 +1,143 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+
+publicWidget.registry.hrRecruitment = publicWidget.Widget.extend({
+    selector : '#hr_recruitment_form',
+
+    events: {
+        'click #apply-btn': '_onClickApplyButton',
+        'focusout #recruitment2' : '_onFocusOutMail',
+        'focusout #recruitment3' : '_onFocusOutPhoneNum',
+        'focusout #recruitment4' : '_onFocusOutLinkedin',
+    },
+
+    init: function () {
+        this._super.apply(this, arguments);
+    },
+
+    willStart() {
+
+        return Promise.all([
+            this._super(),
+        ]);
+    },
+
+    _onClickApplyButton (ev) {
+
+        const linkedin_profile = $('#recruitment4').val();
+        const resume = $('#recruitment6').val();
+
+        if (linkedin_profile.trim() === '' &&
+            resume.trim() === '') {
+
+            $('#recruitment4').attr('required', true);
+            $('#recruitment6').attr('required', true);
+
+        } else {
+
+            $('#recruitment4').attr('required', false);
+            $('#recruitment6').attr('required', false);
+
+        }
+
+    },
+
+    async _onFocusOutPhoneNum (ev) {
+
+        const phone_number = $(ev.currentTarget).val();
+
+        if (!phone_number) {
+            $(ev.currentTarget).removeClass('border-warning');
+            $('#phone-message').removeClass('alert-warning').hide();
+            return;
+        }
+
+        /* This regex matches a general type of phone number that may or may
+         * not include country code, area code, and extension. 
+         * It allows for various separators such as spaces, dots, slashes,
+         * and hyphens.
+         * Here some exemple of valid phone numbers:
+         * +1 (555) 555-5555
+         * 555-555-5555
+         * 555.555.5555
+         * 555/555-5555
+         * (555) 555-5555
+         * +44 20 1234 5678
+         * 12345 67890
+         * [Xs(sXs)]Xs[s(sXs)]XsXsXs[sXsXs]Xs[sXsXs]Xs
+         * */
+        const phone_number_regex = /^(?:\+?\(?\d{1,3}\)?[\s./-]?)?(?:\(\d{1,5}\)|\d{1,5})[\s./-]?\d{1,10}[\s./-]?\d{1,10}[\s./-]?\d{1,10}$/;
+
+        if (!phone_number_regex.test(phone_number)) {
+
+            $('#phone-message').removeClass('alert-warning').hide();
+            $(ev.currentTarget).addClass('border-warning');
+            $('#phone-message').text("Your phone number doesn't match any known standard format. If it's not a mistake you can continue.").addClass('alert-warning').show();
```

Comment:
needs translation with `_t` operator from `import { _t } from "@web/core/l10n/translation";`
```suggestion
            $('#phone-message').text(_t("Your phone number doesn't match any known standard format. If it's not a mistake you can continue.")).addClass('alert-warning').show();
```